numpy/distutils/conv_template.py

```diff
@@ -331,7 +328,10 @@ def main():
     except ValueError:
         e = get_exception()
         raise ValueError("In %s loop at %s" % (file, e))
+    finally:
+        fid.close()
     outfile.write(writestr)
+    outfile.close()
```

Comment:
The trick here is to use:
```
try:
    file = sys.argv[1]
except IndexError:
    fid = contextlib_nullcontext(sys.stdin)
    outfile = contextlib_nullcontext(sys.stdout)
else:
    # as before
```

Then you can use a `with` statement no matter what.

Reply:
`contextlib_nullcontext` is a backport, look around for existing uses of it.